inst/include/distributions/functors/lognormal_lpdf.hpp

```diff
@@ -1,74 +1,94 @@
-#ifndef LOGLogNormal_LPDF 
-#define LOGLogNormal_LPDF 
+#ifndef LOGNORMAL_LPDF
+#define LOGNORMAL_LPDF
 
 #include "density_components_base.hpp"
-#include "../common/fims_vector.hpp"
-#include "../common/def.hpp"
+#include "../../common/fims_vector.hpp"
+#include "../../common/def.hpp"
 
-namespace fims_distributions {
-/**
- * LogNormal Negative Log-Likelihood
- */
-template<typename Type>
-struct LogNormalLPDF : public DensityComponentBase<Type> {
-    fims::Vector<Type> log_sd;
-    fims::Vector<Type> mu;
-    fims::Vector<Type> sd;
-    Type nll = 0.0;
-    //data_indicator<tmbutils::vector<Type> , Type> keep;
-   
+namespace fims_distributions
+{
+    /**
+     * LogNormal Negative Log-Likelihood
+     */
+    template <typename Type>
+    struct LogNormalLPDF : public DensityComponentBase<Type>
+    {
+        fims::Vector<Type> log_sd;
+        fims::Vector<Type> mu;
+        fims::Vector<Type> sd;
+        std::vector<bool> is_na;
+        #ifdef TMB_MODEL
+        ::objective_function<Type> *of;
+        #endif
+        Type nll = 0.0;
+        // data_indicator<tmbutils::vector<Type> , Type> keep;
 
-    LogNormalLPDF() : DistributionComponentBase<Type>() {
+        LogNormalLPDF() : DensityComponentBase<Type>()
+        {
+        }
 
-    }
+        virtual ~LogNormalLPDF() {}
 
-    virtual ~LogNormalLPDF() {}
+        virtual const Type evaluate(const bool &do_log)
+        {
+            this->mu.resize(this->observed_values.size());
+            this->sd.resize(this->observed_values.size());
+            is_na.resize(this->observed_values.size());
+            for (size_t i = 0; i < this->expected_values.size(); i++)
+            {
+                if (this->expected_values.size() == 1)
```

Comment:
I do not understand why this if statement is needed, perhaps I am missing something. If only line 45 was used, then wouldn't mu[0] be correctly defined when i=0? Is this if statement trying to fill a vector with a single value? If so, I do not see how it works, but that could be my misunderstanding.

Reply:
@cmlegault your observation is correct, the conditional statement is not needed. Also, is_na will always be false being initialized on line 36 and used at line 60 without being updated.

Reply:
Yes it's filling a vector with a scalar in a case like a random effect where you have a single mean and SD but a vector of deviations. But it does seem like it should loop over `observed_values.size()` instead of `expected_values.size()` right and combine the two for loops together then right?

Reply:
My logic is that dnorm(observed_values[i], mu[i], sd[i]) needs to work for situations where all three arguments are scalar (e.g. prior), observed_values is a vector but mu is a scalar and sd is a scalar or vector (random effects), and where all three are vectors in the case of data. I think @Cole-Monnahan-NOAA is correct in that the loop should be over observed_values.size().